### PR TITLE
Create API abstract base class

### LINKED ISSUE #19: Refactor lib to allow plugging in own OAuth clients
We're building a free account linking service for Home Assistant. It will allow users to link their third-party accounts via OAuth2 without having to sign up for developer accounts etc.

I've been looking at the different implementations in Home Assistant that could support that, and it seems like the Somfy integration, based on this lib, is a great candidate to be the first implementation.

With the account linking service the client ID and secret will be kept in a cloud service, and all requests for getting access tokens (all grant types) will be routed via it. Authenticated requests to the Somfy API to interact with devices will still be made directly from Home Assistant to Somfy.

But to be able to get it working, we would need to update the `SomfyApi` class to take an instance of a class that implements `get` and `post`. The `get_authorization_url` and `request_token` methods could become helpers.

Would you accept a PR for this lib that accomplishes this? Will also update the Home Assistant side.

btw, are you on the Home Assistant Discord? I couldn't find you.

### PR DESCRIPTION
**Pull request by balloob**
This refactors the SomfyAPI class to allow different implementations for sending `GET` and `POST` requests to the server. 

It's backwards compatible, all tests still pass.

Fixes #19

### CODE REVIEW

File: pymfy/api/somfy_api.py [hunk 1]
```diff
@@ -12,8 +12,56 @@
 SOMFY_REFRESH = 'https://accounts.somfy.com/oauth/oauth/v2/token'
 
 
-class SomfyApi:
-    __slots__ = '_oauth', 'client_id', 'client_secret'
+class AbstractSomfyApi:
```
Comment (tetienne) [inline pymfy/api/somfy_api.py]: I would make it inherit from ABC: https://docs.python.org/3/library/abc.html

File: pymfy/api/somfy_api.py [hunk 2]
```diff
@@ -12,8 +12,56 @@
...
+
+    base_url = BASE_URL
+
+    def get(self, path):
```
Comment (tetienne) [inline pymfy/api/somfy_api.py]: And so add *abstractmethod* decorator.

```python
@abstractmethod
def get(self, path):
   """Fetch a URL from the Somfy API."""
  pass
```

**Comment by balloob** [comment]
Done!

**Comment by tetienne** [comment]
Perfect!

**Pull request merged**